### ISSUE TITLE
Updating a GKE cluster results in a 400

### ISSUE THREAD

**Issue by KaoruDev** [pulumi/pulumi-google-native#418]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details
I'm trying to update a few things on an existing GKE cluster such as:

* `releaseChannel.channel`
* `masterAuthorizedNetworksConfig.cidrBlocks[0].displayName`

but get a `400` response

### Steps to reproduce

1. Create a new GKE cluster with `releaseChannel.channel` set to `STABLE`
2. Modify `releaseChannel.channel` to `REGULAR`
3. Get back an error:

```
Error 400: Must provide an update.: "https://container.googleapis.com/v1/projects/my-project/locations/us-east1/clusters/demo" map[name: demo projectId: my-project]
```

I can update these attributes via `gcloud`

Expected: I am able to update attributes on the cluster.
Actual: `400` return code

**Comment by ColeSiegelTR** [comment]
Got the same issue with different attributes being updated (initialClusterVersion, nodePools)

**Comment by viveklak** [comment]
Fixed in #632
